synapse/http/server.py

```diff
@@ -809,3 +830,24 @@ def finish_request(request: Request):
         request.finish()
     except RuntimeError as e:
         logger.info("Connection disconnected before response was written: %r", e)
+
+
+async def _async_write_json_to_request_in_thread(
```

Comment:
can you move this further up the file (above `_write_json_bytes_to_request`), so that the flow jumps up and down less?